src/txn/txn.c

```diff
@@ -872,11 +885,28 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 					break;
 			}
 
-			if ((updp = ref->page_del->update_list) != NULL)
-				for (; *updp != NULL; ++updp)
+			if ((updp = ref->page_del->update_list) == NULL) {
+				/*
+				 * Publish to ensure we don't let the page be
+				 * evicted and the updates discarded before
+				 * being written.
+				 */
+				WT_PUBLISH(ref->state, previous_state);
+				break;
+			}
+
+			for (; *updp != NULL; ++updp) {
+				if (prepared_transaction) {
+					(*updp)->state = WT_UPDATE_STATE_LOCKED;
 					__wt_timestamp_set(
 					    &(*updp)->timestamp,
 					    &txn->commit_timestamp);
+					(*updp)->state = WT_UPDATE_STATE_READY;
+				} else
+					__wt_timestamp_set(
+					    &(*updp)->timestamp,
+					    &txn->commit_timestamp);
+			}
```

Comment:
Can't we just do:
```
--- ../txn/txn.c        2018-03-20 12:46:42.443256000 -0400
+++ -   2018-03-20 14:13:36.226263126 -0400
@@ -885,17 +885,8 @@
                                        break;
                        }

-                       if ((updp = ref->page_del->update_list) == NULL) {
-                               /*
-                                * Publish to ensure we don't let the page be
-                                * evicted and the updates discarded before
-                                * being written.
-                                */
-                               WT_PUBLISH(ref->state, previous_state);
-                               break;
-                       }
-
-                       for (; *updp != NULL; ++updp) {
+                       for (updp = ref->page_del->update_list;
+                           *updp != NULL; ++updp) {
                                if (prepared_transaction) {
                                        (*updp)->state = WT_UPDATE_STATE_LOCKED;

                                        __wt_timestamp_set(
```

Reply:
On second thought, this is pretty deep indentation -- happy to go with whatever you think.

Reply:
yes keith, the current way is to reduce deep indentation.